sklearn/decomposition/tests/test_pca.py

```diff
@@ -389,6 +390,12 @@ def test_pca_validation():
                                     PCA(n_components, svd_solver=solver)
                                     .fit, data)
 
+    n_components = 1.2
```

Comment:
Actually, can we please test for different solvers?

Reply:
Sure.
It might make sense then to add the checks on individual solvers (see the discussion/comments in the issues).